app/src/main/java/com/example/wnews/views/home/news/NewsProvider.kt

```diff
@@ -0,0 +1,156 @@
+            userAuthProvider.uid
+        )
+        call.enqueue(object : Callback<LikeResponse?> {
+
+            override fun onResponse(
+                call: Call<LikeResponse?>?,
+                response: Response<LikeResponse?>?
+            ) {
+
+                if (response!!.isSuccessful) {
+
+                    newsView.onSuccessResponse()
+
+                } else if (response.errorBody() != null) {
+                    newsView.onFailureResponse(response.message())
+                }
+
+
+            }
+
+            override fun onFailure(call: Call<LikeResponse?>?, t: Throwable?) {
+                newsView.onFailureResponse("")
+            }
+
+        })
+    }
+
+    fun clearArrayList() {
+        arrayListNews = arrayListOf()
+    }
+
+    fun onResponseNews(page: Int) {
+        isLoading = true
+
+        if (page != 1) {
+            newsViewProvider.showProgressBar(isLoading)
+        }
+
+        val call: Call<ListNewsResponse> = RetrofitProvider.newsService.getNews(
+            page,
+            userAuthProvider.token,
+            userAuthProvider.client,
+            userAuthProvider.uid
+        )
+        call.enqueue(object : Callback<ListNewsResponse?> {
+
+            override fun onResponse(
+                call: Call<ListNewsResponse?>?,
+                response: Response<ListNewsResponse?>?
+            ) {
+
+                if (response!!.isSuccessful) {
+
+                    if (page == 1) {
+                        clearArrayList()
+                    }
+
+                    val items = response.body()
+
+
+                    if (!items!!.page.isNullOrEmpty()) {
+                        items.page.forEach { item ->
+
+                            arrayListNews.add(
+                                News(
+                                    item.newsId,
+                                    item.title,
+                                    item.detail,
+                                    getRandomImage(),
+                                    item.like
+                                )
+                            )
+                        }
+                    }
+
+                    newsViewProvider.onSuccessResponse()
+
+                } else if (response.errorBody() != null) {
+                    newsViewProvider.onFailureResponse(response.message())
+                }
+
+                if (page == 1) {
+                    newsViewProvider.initAdapter()
+                }
+
+                newsViewProvider.onNewPageReceived()
+                isLoading = false
+                newsViewProvider.showProgressBar(isLoading)
+
+            }
+
+            override fun onFailure(call: Call<ListNewsResponse?>?, t: Throwable?) {
+                newsViewProvider.onFailureResponse("")
+                isLoading = false
+                newsViewProvider.showProgressBar(isLoading)
+
+            }
+
+        })
+    }
+
+    fun isUserLikeNews(likes: List<Int>): Boolean {
```

Comment:
I would recommend changing this method name to something like `currentUserLikesNews`